gradle.lockfile

```diff
@@ -0,0 +1,239 @@
+# This is a Gradle generated file for dependency locking.
+# Manual edits can break the build and are not advised.
+# This file is expected to be part of source control.
```

Comment:
So does this just get created/modified automatically? Do we need to change anything about the developer workflow?

Reply:
After changing a dep, developers should run (for example) `./gradlew dependencies --write-locks` ([reference](https://docs.gradle.org/current/userguide/dependency_locking.html#lock_all_configurations_in_one_build_execution))

Reply:
Can we enforce that somehow, similar to needing to run spotlessApply before committing?

Reply:
See [here](https://docs.gradle.org/current/userguide/dependency_locking.html#running_a_build_with_lock_state_present), we could change the mode to strict but I don't _quite_ understand the difference

Reply:
I suggest we start with strict, and see if that causes us any problems. Though: I don't think strict mode is what @wchamber was talking about. I think we'd have to enforce that developers call `--write-locks` through PR review and such.

Reply:
Working on making things clearer.
FWIW, here's the output when trying to run `unitTest` with some added unlocked dependencies:
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Could not resolve all files for configuration ':compileClasspath'.
   > Resolved 'ch.qos.logback.contrib:logback-json-classic:0.1.5' which is not part of the dependency lock state
   > Resolved 'com.google.cloud:google-cloud-logging:1.101.2' which is not part of the dependency lock state
   > Resolved 'com.google.api.grpc:proto-google-cloud-logging-v2:0.84.2' which is not part of the dependency lock state
   > Resolved 'ch.qos.logback.contrib:logback-json-core:0.1.5' which is not part of the dependency lock state
   > Resolved 'org.springframework.cloud:spring-cloud-gcp-logging:1.2.5.RELEASE' which is not part of the dependency lock state
   > Did not resolve 'com.google.auto.value:auto-value-annotations:1.7.2' which has been forced / substituted to a different version: '1.7.4'
   > Resolved 'org.springframework.cloud:spring-cloud-gcp-starter-logging:1.2.5.RELEASE' which is not part of the dependency lock state
   > Resolved 'org.checkerframework:checker-compat-qual:2.5.5' which is not part of the dependency lock state
   > Resolved 'com.google.cloud:google-cloud-logging-logback:0.118.0-alpha' which is not part of the dependency lock state
```

Reply:
I'm leaning towards not enabling strict mode.
The main reason is that all the configurations we care about are already locked, which means that tools looking for synthetic configurations that don't actually exits get an error that there's no lockstate for them, rather than just seeing that the configurations don't exist. I discovered this with the [versions plugin](https://github.com/ben-manes/gradle-versions-plugin)